src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Date.O.cs

```diff
@@ -51,46 +51,52 @@ private static bool TryFormatDateTimeO(DateTime value, TimeSpan offset, Span<byt
             // Hoist most of the bounds checks on buffer.
             { _ = destination[MinimumBytesNeeded - 1]; }
 
-            FormattingHelpers.WriteFourDecimalDigits((uint)value.Year, destination, 0);
+            value.GetDate(out int year, out int month, out int day);
+            value.GetTimePrecise(out int hour, out int minute, out int second, out int ticks);
+
+            FormattingHelpers.WriteFourDecimalDigits((uint)year, destination, 0);
             destination[4] = Utf8Constants.Minus;
 
-            FormattingHelpers.WriteTwoDecimalDigits((uint)value.Month, destination, 5);
+            FormattingHelpers.WriteTwoDecimalDigits((uint)month, destination, 5);
             destination[7] = Utf8Constants.Minus;
 
-            FormattingHelpers.WriteTwoDecimalDigits((uint)value.Day, destination, 8);
+            FormattingHelpers.WriteTwoDecimalDigits((uint)day, destination, 8);
             destination[10] = TimeMarker;
 
-            FormattingHelpers.WriteTwoDecimalDigits((uint)value.Hour, destination, 11);
+            FormattingHelpers.WriteTwoDecimalDigits((uint)hour, destination, 11);
             destination[13] = Utf8Constants.Colon;
 
-            FormattingHelpers.WriteTwoDecimalDigits((uint)value.Minute, destination, 14);
+            FormattingHelpers.WriteTwoDecimalDigits((uint)minute, destination, 14);
             destination[16] = Utf8Constants.Colon;
 
-            FormattingHelpers.WriteTwoDecimalDigits((uint)value.Second, destination, 17);
+            FormattingHelpers.WriteTwoDecimalDigits((uint)second, destination, 17);
             destination[19] = Utf8Constants.Period;
 
-            FormattingHelpers.WriteDigits((uint)((ulong)value.Ticks % (ulong)TimeSpan.TicksPerSecond), destination.Slice(20, 7));
+            FormattingHelpers.WriteDigits((uint)ticks, destination.Slice(20, 7));
```

Comment:
The `% (ulong)TimeSpan.TicksPerSecond` isn't necessary because we know it's less than that limit?

Reply:
(Reading further I see the answer is "yes")